project-koku/koku
# [COST-4178] - (OCP) determine metadata during csv creation

## Jira Ticket

[COST-4178](https://issues.redhat.com/browse/COST-4178)

## Description

This change will create the necessary s3 metadata when the daily csv files are created. This metadata is passed to the parquet processor. Several areas are updated to use pathlib Path instead of strings. This makes grabbing the correct metadata easier.

## Testing

1. `make delete-test-customer-data; make delete-testing; make delete-trino-data; make delete-redis-cache`
2. generate nise data (requires latest version): 
```
nise -lll report ocp --insights-upload testing/data/insights_local --ocp-cluster-id my-ocp-cluster-1 -s 2023-09-01T00 -e 2023-09-06T12 --daily-reports
```
3. `make create-test-customer`
4. check minio and see the metadata for the each file has a start-date matching the date of the file and the `num hours` matches how many hours worth of data are in the files
5. create nise data which overlaps the first data set:
```
nise -lll report ocp --insights-upload testing/data/insights_local --ocp-cluster-id my-ocp-cluster-1 -s 2023-09-06 -e 2023-09-07 --daily-reports
```
6. download: http://localhost:5042/api/cost-management/v1/download/
7. check minio, see a single file for the day on which data overlaps (with the above commands, that day is `2023-09-06`)

## Notes

...
